lib/Sema/ConstraintSystem.cpp

```diff
@@ -2729,6 +2729,154 @@ static void diagnoseOperatorAmbiguity(ConstraintSystem &cs,
               llvm::join(parameters, ", "));
 }
 
+std::string swift::describeGenericType(ValueDecl *GP, bool includeName) {
+  if (!GP)
+    return "";
+
+  Decl *parent = nullptr;
+  if (auto *AT = dyn_cast<AssociatedTypeDecl>(GP)) {
+    parent = AT->getProtocol();
+  } else {
+    auto *dc = GP->getDeclContext();
+    parent = dc->getInnermostDeclarationDeclContext();
+  }
+
+  if (!parent)
+    return "";
+
+  llvm::SmallString<64> result;
+  llvm::raw_svector_ostream OS(result);
+
+  OS << Decl::getDescriptiveKindName(GP->getDescriptiveKind());
+
+  if (includeName && GP->hasName())
+    OS << " '" << GP->getBaseName() << "'";
+
+  OS << " of ";
+  OS << Decl::getDescriptiveKindName(parent->getDescriptiveKind());
+  if (auto *decl = dyn_cast<ValueDecl>(parent)) {
+    if (decl->hasName())
+      OS << " '" << decl->getFullName() << "'";
+  }
+
+  return OS.str();
+}
+
+/// Special handling of conflicts associated with generic arguments.
+///
+/// func foo<T>(_: T, _: T) {}
+/// func bar(x: Int, y: Float) {
+///   foo(x, y)
+/// }
+///
+/// It's done by first retrieving all generic parameters from each solution,
+/// filtering boundings into distrinct set and diagnosing any differences.
```

Comment:
Typos: boundings->bindings and distrinct->distinct?

Reply:
Fixed, thank you!